cartography/intel/okta/oktaintel.py

```diff
@@ -0,0 +1,1225 @@
+    else:
+        group_props["windows_domain_qualified_name"] = None
+
+    if okta_group.profile.externalId:
+        group_props["external_id"] = okta_group.profile.externalId
+    else:
+        group_props["external_id"] = None
+
+    return group_props
+
+
+def _sync_okta_groups(neo4_session, okta_org_id, okta_update_tag):
+    """
+    Synchronize okta groups
+    :param neo4_session: session with the Neo4j server
+    :param okta_org_id: okta organization id
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+    logger.debug("Syncing Okta groups")
+    api_client = _create_api_client(okta_org_id, "/api/v1/groups")
+
+    data = _get_okta_groups(api_client)
+    _load_okta_groups(neo4_session, okta_org_id, data, okta_update_tag)
+
+    _load_okta_group_membership(neo4_session, api_client, okta_org_id, okta_update_tag)
+
+
+def _create_okta_organization(neo4j_session, organization, okta_update_tag):
+    """
+    Create Okta organization in the graph
+    :param neo4_session: session with the Neo4j server
+    :param organization: okta organization id
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+    ingest = """
+    MERGE (org:OktaOrganization{id: {ORG_NAME}})
+    ON CREATE SET org.name = org.id, org.firstseen = timestamp()
+    SET org.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest,
+        ORG_NAME=organization,
+        okta_update_tag=okta_update_tag,
+    )
+
+
+def _load_okta_groups(neo4j_session, okta_org_id, group_list, okta_update_tag):
+    """
+    Add okta groups to the graph
+    :param neo4j_session: session with the Neo4j server
+    :param okta_org_id: okta organization id
+    :param group_list: group of list
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+    ingest_statement = """
+    MATCH (org:OktaOrganization{id: {ORG_ID}})
+    WITH org
+    UNWIND {GROUP_LIST} as group_data
+    MERGE (new_group:OktaGroup{id: group_data.id})
+    ON CREATE SET new_group.firstseen = timestamp()
+    SET new_group.name = group_data.name,
+    new_group.description = group_data.description,
+    new_group.sam_account_name = group_data.sam_account_name,
+    new_group.dn = group_data.dn,
+    new_group.windows_domain_qualified_name = group_data.windows_domain_qualified_name,
+    new_group.external_id = group_data.external_id,
+    new_group.lastupdated = {okta_update_tag}
+    WITH new_group, org
+    MERGE (org)-[org_r:RESOURCE]->(new_group)
+    ON CREATE SET org_r.firstseen = timestamp()
+    SET org_r.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest_statement,
+        ORG_ID=okta_org_id,
+        GROUP_LIST=group_list,
+        okta_update_tag=okta_update_tag,
+    )
+
+
+def _get_okta_groups_id_from_graph(neo4j_session, okta_org_id):
```

Comment:
Rather than query the graph for data we had just loaded into it, I think we should just use the groups object returned from a previous call.  This will save IO and network calls.  

Instead, I recommend using the `attach` pattern used here: https://github.com/lyft/cartography/blob/master/cartography/intel/aws/rds.py#L108

Reply:
Removed the call when we already have the group list. The function is needed when we map roles and applications.